csharp/rust/src/lib.rs

```diff
@@ -1,284 +1,472 @@
+        Ok(d) => d,
+        Err(e) => {
+            logger_core::log(Error, 
+                "csharp_ffi",
+                format!("Error in string transformation: {:?}", e.to_string()),
+            );
+            return match e {
+                Utf8OrEmptyError::Utf8Error(e) => {
+                    CommandResult::new_error(helpers::to_cstr_ptr_or_null(e.to_string().as_str()))
+                }
+                Utf8OrEmptyError::Empty => CommandResult::new_error(helpers::to_cstr_ptr_or_null(
+                    "Null value passed for host",
+                )),
+            };
+        }
+    };
+    let cmd = match in_request_type.get_command() {
+        None => {
+            logger_core::log(Error, 
+                "csharp_ffi",
+                "Error in csharp_command called with unknown request type",
+            );
+            return CommandResult::new_error(helpers::to_cstr_ptr_or_null("Unknown request type"));
+        }
+        Some(d) => d,
+    };
+    let callback = in_callback;
+    let callback_data = in_callback_data as usize;
+
+    let ffi_handle = unsafe { Box::leak(Box::from_raw(in_client_ptr as *mut FFIHandle)) };
+    let handle = ffi_handle.handle.clone();
+    let routing_info = if in_routing_info.is_null() {
+        None
+    } else {
+        Some(unsafe {
+            match (*in_routing_info).to_redis() {
+                Ok(d) => d,
+                Err(e) => {
+                    logger_core::log(Error, 
+                        "csharp_ffi",
+                        format!(
+                            "Error while parsing route in string transformation: {:?}",
+                            e.to_string()
+                        ),
+                    );
+                    return match e {
+                        Utf8OrEmptyError::Utf8Error(e) => CommandResult::new_error(
+                            helpers::to_cstr_ptr_or_null(e.to_string().as_str()),
+                        ),
+                        Utf8OrEmptyError::Empty => {
+                            CommandResult::new_error(helpers::to_cstr_ptr_or_null(
+                                "Routing info incomplete, null value passed in string",
+                            ))
+                        }
+                    };
+                }
+            }
+        })
+    };
+    ffi_handle.runtime.spawn(async move {
+        logger_core::log(Trace, "csharp_ffi", "Entered command task with");
+        let data: redis::Value = match handle.command(cmd, args, routing_info).await {
+            Ok(d) => d,
+            Err(e) => {
+                logger_core::log(Error, 
+                    "csharp_ffi",
+                    format!(
+                        "Error handling command in task of csharp_command: {:?}",
+                        e.to_string()
+                    ),
+                );
+                let value = Value::simple_string_with_null(e.to_string().as_str());
+                match catch_unwind(|| unsafe {
```

Comment:
I propose adding panic handling in another PR. Need to handle those in client creation too

Reply:
This isn’t about panic handling per se, but rather about catching unexpected exceptions that may slip through from the .NET side. While, in theory, such exceptions shouldn't occur, I’ve added this as an extra safety measure to prevent potential program faults. It’s a defensive safeguard to ensure stability, even in edge cases where something goes wrong unexpectedly.